lib/microformats/format_parser.rb

```diff
@@ -7,6 +7,7 @@ def parse(element, base: nil, element_type: nil, format_class_array: [], backcom
 
       @properties = {}
       @children = []
+      @seen_types = {}
```

Comment:
What does `@seen_types` refer to?

Reply:
this is an internal variable to note all the prefixes seen in a microformat.  The change to when to add implied url or name requires somewhere to track this

Reply:
also, yes, totally cool with you documenting and updating

Reply:
just pushed updated test set as well

Reply:
> this is an internal variable to note all the prefixes seen in a microformat. The change to when to add implied url or name requires somewhere to track this

Ah, okay, so it's part of the implied properties bits. I need to do a bit more reading on the wiki to fully understand that.

Would a better name for this be something like `matched_prefixes` or `found_prefixes` or something similar…? Or is "type" the appropriate nomenclature for `h-`, `p-`, `u-`, etc. ?

Reply:
found_prefixes makes more sense certainly.  

Using set / subset works for me.
I am not good at all the fancier ways of doing things in ruby, so i'll take any suggestions certainly.  That looks like a much cleaner way to go

Reply:
> Ah, okay, so it's part of the implied properties bits. I need to do a bit more reading on the wiki to fully understand that.

It changed 

> if no explicit "name" property 

to 

> if no explicit "name" property, and no other p-* or e-* properties, and no nested microformats, 

and similarly  for url 

> if no explicit "url" property,

to 

> if no explicit "url" property, and no other explicit u-* properties, and no nested microformats,

Reply:
> I am not good at all the fancier ways of doing things in ruby, so i'll take any suggestions certainly. That looks like a much cleaner way to go

Not to worry! `Set` is pretty new to me, too. I happened upon it while looking at ways we might clean up `@seen_types`. I can take a swing at the conversion and if it works out will push up a commit.